crates/ruff_python_parser/src/lexer.rs

```diff
@@ -508,6 +514,138 @@ impl<'source> Lexer<'source> {
+                }
+                '\\' => {
+                    self.cursor.bump();
+                    if matches!(self.cursor.first(), '{' | '}') {
+                        // Don't consume `{` or `}` as we want them to be consumed as tokens.
+                        break self.offset() - TextSize::new(1);
+                    } else if !context.is_raw_string {
+                        if self.cursor.first() == 'N' && self.cursor.second() == '{' {
+                            self.cursor.bump();
+                            self.cursor.bump();
+                            in_named_unicode = true;
+                        }
+                    }
+                }
+                quote @ ('\'' | '"') => {
+                    if quote == context.quote_char() {
+                        match context.quote_size() {
+                            StringQuoteSize::Single => {
+                                break self.offset();
+                            }
+                            StringQuoteSize::Triple => {
+                                let mut remaining = self.cursor.rest().chars().clone();
+                                if remaining.next() == Some(quote)
+                                    && remaining.next() == Some(quote)
+                                {
+                                    break self.offset();
+                                }
+                            }
+                        }
+                    }
+                }
+                '{' => {
+                    if self.cursor.second() == '{' {
+                        self.cursor.bump();
+                        self.cursor.bump();
+                    } else {
+                        break self.offset();
+                    }
+                }
+                '}' => {
+                    if in_named_unicode {
+                        in_named_unicode = false;
+                        self.cursor.bump();
+                    } else if self.cursor.second() == '}' && !context.is_in_format_spec() {
+                        self.cursor.bump();
+                        self.cursor.bump();
+                    } else {
+                        break self.offset();
+                    }
+                }
+                _ => {
+                    self.cursor.bump();
+                }
+            }
+        };
+
+        if value_start == value_end {
+            return Ok(None);
+        }
+
+        let value = self.source[TextRange::new(value_start, value_end)]
+            .to_string()
+            .replace("{{", "{")
+            .replace("}}", "}");
+        Ok(Some(Tok::FStringMiddle(value)))
+    }
+
+    fn try_lex_fstring_end(&mut self) -> Option<Tok> {
+        // SAFETY: Safe because the function is only called when `self.fstring_stack` is not empty.
+        let context = self.fstring_stack.last().unwrap();
+
+        if let quote @ ('\'' | '"') = self.cursor.first() {
+            if quote == context.quote_char() {
```

Comment:
```suggestion
        if self.cursor.first() == context.quote_char() {
```

Reply:
We actually need the `quote` for the lookaheads if it's a triple quoted string.

Reply:
Could we use `context.quote_char`?